options.go

```diff
@@ -222,6 +227,20 @@ func (opts *Options) Initialize() error {
 	return nil
 }
 
+// Validate checks that all required fields are set, and no options are incorrectly
+// configured.
+func (opts *Options) Validate() error {
```

Comment:
nit: It might be nice to have this be a []error instead so you can see all the errors that validate returns. Don't feel strongly about this though.

Reply:
I would make a `type ValidationError []error` for this. I can copy that code over: https://godoc.org/code.cloudfoundry.org/bbs/models#ValidationError

I thought about adding it, but didn't because KISS and we could add it later.

Reply:
That's reasonable. It feels like a great to have feature but no where close to a requirement.